homeassistant/components/roborock/config_flow.py

```diff
@@ -137,6 +139,24 @@ async def async_step_code(
             errors=errors,
         )
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle a flow started by a dhcp discovery."""
+        roborock_entries = self.hass.config_entries.async_entries(DOMAIN)
+        device_registry = dr.async_get(self.hass)
+        for entry in roborock_entries:
```

Comment:
I was reviewing other integrations that do this and they have a nice simple way; https://github.com/home-assistant/core/blob/ddd67a7e58c2a13d6126e8542ac8eaa74cbf6a25/homeassistant/components/ring/config_flow.py#L80 using `device_registry.async_get_device` to do this in a single call

Reply:
with connections or identifiers? I'm not sure how to do identifier for domain without having the config entry id, but I guess I could call async_get_devices for the mac and then check if is is a roborock device?

Reply:
Thoughts on the latest?

Reply:
Seems to be what this does: https://github.com/home-assistant/core/blob/ddd67a7e58c2a13d6126e8542ac8eaa74cbf6a25/homeassistant/components/dhcp/__init__.py#L262

```
        if device := dev_reg.async_get_device(
            connections={(CONNECTION_NETWORK_MAC, formatted_mac)}
        ):
            for entry_id in device.config_entries:
                if (
                    entry := self.hass.config_entries.async_get_entry(entry_id)
                ) and entry.domain
```

Not sure that is any shorter than what is here.

So connection[1] is supposed to already be formatted with `format_mac` and if its not, then that needs to be repaired separately.

If you keep it this way it could be shortened this way:
```suggestion
            mac = dr.format_mac(discovery_info.macaddress)
            for device in device_entries:
                for connection in device.connections:
                    if connection == (CONNECTION_NETWORK_MAC, mac):
```